crates/fuel-core/src/graphql_api/database.rs

```diff
@@ -226,6 +230,22 @@ where
     }
 }
 
+impl StorageSize<BlobData> for ReadView {
+    fn size_of_value(&self, key: &BlobId) -> Result<Option<usize>, Self::Error> {
+        self.on_chain.size_of_value(key)
+    }
+}
+
+impl StorageRead<BlobData> for ReadView {
+    fn read(&self, key: &BlobId, buf: &mut [u8]) -> Result<Option<usize>, Self::Error> {
+        self.on_chain.read(key, buf)
```

Comment:
The Blob is being read from the on-chain view, and will be subsequently used to estimate and verify the predicate. 
I am making the assumption here that blobs are immutable, but won't necessarily be persisted on-chain. In particular, I am assuming that we are referring to EIP-4844 blobs. 

If this assumption holds, are predicates still stateless? E.g. what happens if you re-execute a predicate that loads a blob whose id is hardcoded in the predicate code?

Reply:
(overall this might not be a problem since we finalize the state after 7 days, but worth mentioning)

Reply:
These blobs are not related to the Ethereum=) It is our blobs https://github.com/FuelLabs/fuel-specs/blob/master/src/tx-format/transaction.md#transactionblob

Basically any bytes that we store on chain.

Reply:
Thanks :) Also my concerns were answered by your comment here: https://github.com/FuelLabs/fuel-vm/pull/848#issuecomment-2395118811.